core/src/edu/cornell/gdiac/main/view/CutsceneMode.java

```diff
@@ -0,0 +1,308 @@
+import com.badlogic.gdx.utils.viewport.ScreenViewport;
+import com.badlogic.gdx.utils.viewport.Viewport;
+import edu.cornell.gdiac.assets.AssetDirectory;
+import edu.cornell.gdiac.main.controller.InputController;
+import edu.cornell.gdiac.util.ScreenListener;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Carousel tutorial mode
+ */
+public class CutsceneMode implements Screen {
+  // TODO: (Nice to have) Add index indicator
+
+  /**
+   * Exit code for dismissing and leaving this mode. Triggered either after last slide or on skip.
+   */
+  public static final int EXIT_LEAVE = 0;
+
+  /**
+   * Need an ongoing reference to the asset directory
+   */
+  protected AssetDirectory directory;
+
+  /**
+   * The pixel font used in the game
+   */
+  private BitmapFont pixelFont;
+
+  /**
+   * Listener that will update the player mode when we are done
+   */
+  private ScreenListener listener;
+
+  /**
+   * Reference to GameCanvas created by the root
+   */
+  private GameCanvas canvas;
+
+  /**
+   * Whether this is an active controller
+   */
+  private boolean active;
+
+  /**
+   * Background textures used in this carousel
+   */
+  private List<Texture> textureList;
+
+  /**
+   * Current texture/slide index
+   */
+  private int textureIndex;
+
+  /**
+   * A reference to the input controller
+   */
+  private InputController inputController;
+
+  /**
+   * Menu containing the buttons
+   */
+  private ButtonsMenu buttonsMenu;
+
+
+  public CutsceneMode(GameCanvas canvas) {
+    active = false;
+    this.canvas = canvas;
+    this.textureList = new ArrayList<>();
+
+    // Initialize buttons menu
+    Batch batch = new SpriteBatch();
+    buttonsMenu = new ButtonsMenu(new ScreenViewport(), batch);
+    buttonsMenu.getViewport().update(Gdx.graphics.getWidth(), Gdx.graphics.getHeight(), true);
+  }
+
+  public void init(AssetDirectory directory) {
+    this.directory = directory;
+    textureIndex = 0;
+    this.pixelFont = directory.getEntry("display", BitmapFont.class);
+
+    // Populate textures list (each is itself a page)
+    textureList.add(directory.getEntry("cutscene0", Texture.class));
+    textureList.add(directory.getEntry("cutscene1", Texture.class));
+    textureList.add(directory.getEntry("cutscene2", Texture.class));
+
+    // Set up input controller
+    this.inputController = InputController.getInstance();
+    Gdx.input.setInputProcessor(inputController.getInputProcessor());
+
+    buttonsMenu.init();
+    buttonsMenu.show();
+
+    // Finally, activate the screen
+    active = true;
+  }
+
+  @Override
+  public void show() {
+    active = true;
+  }
+
+  @Override
+  public void render(float delta) {
+    if (!active) {
+      return;
+    }
+
+    // -------- Update ---------
+
+    // Read input from controller
+    assert inputController != null;
+    inputController.readInput(null, null);
+
+    // Handle inputs
+    if (inputController.didJustPressLeft()) {
```

Comment:
Using `didJustPress` methods are inconsistent and only occasionally recognize input. Instead, you should use event-driven input with call-back functions to make sure things are consistent. See the level selector screen as an example.